src/main/java/com/microsoft/azure/functions/binding/RpcComplexDataSources.java

```diff
@@ -81,7 +81,8 @@
         super(name, null, HTTP_DATA_OPERATIONS);
         this.httpPayload = value;
         this.bodyDataSource = BindingDataStore.rpcSourceFromTypedData(null, this.httpPayload.getBody());
-        this.fields = Arrays.asList(this.httpPayload.getHeadersMap(), this.httpPayload.getQueryMap(), this.httpPayload.getParamsMap());
+        this.fields = Arrays.asList(this.httpPayload.getQueryMap(), this.httpPayload.getParamsMap());
+        this.httpHeaders = this.httpPayload.getHeadersMap();
```

Comment:
Remove the header from `this.fields` is not enough, you also need to make sure the metadata is also resolved by name just like [the one here](https://github.com/pragnagopa/azure-functions-java-worker/blob/6499f80522dd5c5b7ccf085fafc4450d120ceaa1/src/main/java/com/microsoft/azure/functions/binding/RpcComplexDataSources.java#L91).

The method above is used to provide the value when developers are using `@Bingding("header-name")` annotation.

Reply:
Let me know what is the right way to fix this. Can you point me to documentation?

Reply:
Let's see whether we need to change the HTTP header at all. If not, there is no need to modify this.